pkg/remotewriteutils/util.go

```diff
@@ -41,5 +41,12 @@ func FetchPrometheusList(ctx context.Context, r *ResourceWrapper, rw *pmov1alpha
 		return nil, microerror.Maskf(errorFetchingPrometheus, "Could not fetch Prometheus with label selector %#q", rw.Spec.ClusterSelector.String())
 	}
 
+	for i, p := range prometheusList.Items {
```

Comment:
Would it not be easier to change the label selector to skip the agent using the not operator and this label app.kubernetes.io/name: prometheus-agent ?

Reply:
it's clusterSelector and not prometheusSelector

Reply:
Well that's how w ecall it but it's used to select prometheus is it not?

Reply:
```go
prometheusList, err := r.PrometheusClient.
		MonitoringV1().
		Prometheuses(metav1.NamespaceAll).
		List(ctx, metav1.ListOptions{LabelSelector: selector.String()})
```

Reply:
Hmmm, let me check!

Reply:
LabelSelecto contains a MatchLabels and MatchExpressions :)

Reply:
So we can add a MatchExpressions Not Equal app.kubernetes.io/name: prometheus-agent

Reply:
let's try

Reply:
works ✅

Reply:
can you take a look?

Reply:
Nice